docs/local-development.md

```diff
@@ -0,0 +1,157 @@
+# Local Development Setup Guide
+
+This guide will help you set up a local development environment for the Restarters.net project using Docker.
+
+- [Local Development Setup Guide](#local-development-setup-guide)
+  - [Prerequisites](#prerequisites)
+  - [Setup Steps](#setup-steps)
+    - [1. Clone the Repository](#1-clone-the-repository)
+    - [2. Environment Configuration](#2-environment-configuration)
+    - [3. Starting the Development Environment](#3-starting-the-development-environment)
+    - [4. Initial Setup](#4-initial-setup)
+    - [5. Accessing the Application](#5-accessing-the-application)
+  - [Common Tasks](#common-tasks)
+    - [Running Commands in the Core Application Container](#running-commands-in-the-core-application-container)
+    - [Stopping the Environment](#stopping-the-environment)
+    - [Rebuilding Containers](#rebuilding-containers)
+****
+
+## Prerequisites
+
+Before you begin, ensure you have the following installed:
+
+- [Docker](https://docs.docker.com/get-docker/) (newer versions already include Docker Compose)
+  - [Docker Compose](https://docs.docker.com/compose/install/) (only needed for older Docker versions)
+- [Task](https://taskfile.dev/installation/)
+
+> [!NOTE]
+> The `Taskfile` will automatically detect your Docker version and use the appropriate command (`docker-compose` or `docker compose`).
+
+## Setup Steps
+
+### 1. Clone the Repository
+
+```bash
+git clone https://github.com/TheRestartProject/restarters.net.git
+cd restarters.net
+```
+
+### 2. Environment Configuration
+
+Copy the example environment file:
+
+```bash
+cp .env.example .env
+```
+
+You may need to adjust the following settings in your `.env` file:
+- The `APP_KEY` will be auto-generated during setup
+> [!NOTE]
+> This is only required if another service is already running on the port `8001`
+- Set `APP_URL` to `http://localhost:<AVAILABLE_PORT>`
+  - You will need to change the port in the `docker-compose.yml` file to match the port you are using.
+
+### 3. Starting the Development Environment
+
+The development environment includes several optional components that can be enabled:
+
+- **Core**: Restarters web application and MySQL database
+- **Debug Tools**: Adds phpMyAdmin and Mailhog for debugging and testing
+- **Discourse**: Adds Discourse forum with its required services (PostgreSQL, Redis, Sidekiq) 
+
+The project includes a Taskfile that provides convenient commands for managing Docker:
+
+```bash
+# Start the core environment (app + database only)
+task docker:up-core
+
+# Start with debug tools (phpMyAdmin, Mailhog)
+task docker:up-debug
+
+# Start with Discourse integration
+task docker:up-discourse
+
+# Start all services
+task docker:up-all
```

Comment:
This is better, thank you.  But it's still not clear what the difference is between `up-all` and `up-discourse`.

Reply:
Within the documentation or the task description, or both?

https://github.com/iFixit/restarters/blob/d790c85ce17c3955346faf8ca9f48652a5f8dbd1/Taskfile.yml#L37-L38

I am not really sure what would be the best solution for this.

Reply:
Within the documentation.  Think of it this way: what containers does `all` start that `discourse` doesn't?  How would I know that?

Reply:
Addressed in 3ce4fdd5f6d92fed39a229c152700ffc28cc7041